src/rmarkdown/knit.ts

```diff
@@ -149,20 +149,26 @@ export class RMarkdownKnitManager extends RMarkdownManager {
 
 	// alters the working directory for evaluating chunks
 	public setKnitDir(): void {
-		const currentDocumentWorkspace = vscode.workspace.getWorkspaceFolder(vscode.window.activeTextEditor.document.uri);
+		let currentDocumentWorkspacePath: string = null;
```

Comment:
I should have made it more clear: untitled document and document outside any workspace folder will return `undefined` workspace folder. We should handle both cases separately.

Reply:
Maybe we could provide QuickPickItem accordingly:

* If it has a workspace folder, then add a QuickPickItem for it.
* If it has a filesystem folder, then add a QuickPickItem for it.